mimic/model/nova_objects.py

```diff
@@ -714,7 +714,7 @@ def request_creation(self, creation_http_request, creation_json,
             behavior = self.create_behavior_registry.behavior_for_attributes({
                 "tenant_id": self.tenant_id,
                 "server_name": creation_json["server"]["name"],
-                "metadata": creation_json["server"].get("metadata")
+                "metadata": creation_json["server"].get("metadata", {})
```

Comment:
Note that this was added in because if metadata was None before, using behaviors seemed to fail (unit tests failed).  This came up as a result of removing the test-specific `create_server` function which always provided a default empty dictionary for metadata.